File: javascript/algorithms/sort/bubble_sort.js
// O(n^2) bubble sort implementation

function swap(array, index1, index2) {
    let temp = array[index1];
    array[index1] = array[index2];
    array[index2] = temp;
}

function bubble_sort(array) {
    for (let i = 0; i < array.length; i++) {
        for (let j = 0; j <= i; j++) {
            if (array[i] < array[j]) {
                swap(array, i, j);
            }
        }
    }
    console.log(array);
    return array;
}

bubble_sort([6,3,2,1,4,5]); // [1,2,3,4,5,6]

File: cpp/algorithms/sort/bubble_sort.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>


using namespace std;

int main() {

    return 0;
}

int bubble_sort(int array []) {
    int n = array.size();

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n-1; j++) {
            if (array[j] > array[j+1]) {
                swap(array[j], array[j+1]);
            }
        }
    }
}

int binary_search() {
    int a = 0, b = n-1;
    while(a<= b) {
        int k = (a+b)/2;
        if (array[k] == x) {

        }
        if (array[k] > x) b = k-1;
        else a = k+1;
    }
}